tests/test_client.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+import requests
+
+from tap_wonolo.client import WonoloStream, is_fatal_code
+
+
+@pytest.mark.parametrize('status_code', [400, 401, 403, 404, 429,
+                                         pytest.param(500, marks=pytest.mark.xfail),
+                                         pytest.param(502, marks=pytest.mark.xfail),
+                                         pytest.param(503, marks=pytest.mark.xfail),
+                                         pytest.param(504, marks=pytest.mark.xfail)])
+def test_is_fatal_code(status_code):
```

Comment:
seems like we want two tests now, one for HTTPError and one for RequestException (with a None response?)

Reply:
@bobzoller it's not possible for a RequestException to have a None reponse for this test. On ly `HTTPErrors` get routed to it, which are guaranteed to have a non-none Response object with a non-none status_code

Reply:
@bobzoller misread your comment - I think you're advocating for a _different_ test, yeah? It wouldn't specifically test this function though, since this function only gets called by HTTPErrors

Reply:
talked in person, this is hard and we're going to skip it